src/plugins/advanced_settings/public/management_app/advanced_settings.tsx

```diff
@@ -198,6 +198,7 @@ export class AdvancedSettings extends Component<AdvancedSettingsProps, AdvancedS
         });
       })
       .filter((c) => !c.readOnly)
+      .filter((c) => !c.isCustom) // hide any settings that aren't explicitly registered by enabled plugins.
```

Comment:
We already have a flag that we can use to filter out any settings that aren't explicitly registered during Kibana setup and start lifecycles